### PR TITLE
Add multicast subcommand   (issueのレビュー用)

### PR DESCRIPTION
**Pull request by yushoyamaguchi**
## Proposal / RFE
Currently, for creating multicast groups and registering subscribers, it is necessary to enter the cilium-agent pod on each node and use the `cilium-dbg` command. 
This is fine but does not scale when we need to use multicast with many nodes in the cluster.
https://github.com/cilium/cilium/pull/29469
https://github.com/cilium/cilium/pull/31355
https://github.com/cilium/cilium/pull/32612
For example, when using multicast application such as ROS 2 and DDS as distributed application, we have many nodes need to join the multicast group to establish the endpoint discovery in application layer.

Therefore, I would like to propose implementing commands in cilium-cli that creates a specified address group inside all cilium-agents and adds all other nodes to that group.
Additionally, I would like to implement commands for deletion and status checking.
https://github.com/cilium/cilium/pull/32612#discussion_r1605542681

### Suggested command examples
These are just examples and we hope to change them for the better through the discussion.
```
#add all nodes as subscriber of 239.1.1.1 in all cilium-agent pods
>cilium multicast group 239.1.1.1 add all
```
```
#delete 239.1.1.1 group in all nodes
>cilium multicast group 239.1.1.1 delete all
```
```
#show list of nodes which have matched multicast group 
>cilium multicast group 239.1.1.1 list
nodeA , nodeB , nodeD
``` 
```
#show all lists of multicast group
cilium multicast list --all 
```

### Furthermore
In the future, it would be beneficial to introduce commands like the following:
```
#add the specified node as subscriber of 239.1.1.1 in all cilium-agent pod
>cilium multicast group 239.1.1.1 add <node-name or node IP address>
```
```
#remove the specified node from 239.1.1.1 group in all cilium-agent pod
>cilium multicast group 239.1.1.1 delete <node-name or node IP address>
```
```
#add all nodes as subscriber of 239.1.1.1 by cilium-dbg
>cilium-dbg bpf multicast subscriber add 239.255.0.1 10.244.1.86
```

### CODE REVIEW

**Comment by fujitatomoya** [comment]
some comments, but overall lgtm. this does not block creating issue in mainline.

- i would add the use cases with multicast application such as ROS 2 and DDS as distributed application. that would be easier to see what we need to do so. currently it is clear that we cannot dynamically handle the multicast configuration when the node joins. even with this constraints, we have many nodes need to join the multicast group to establish the endpoint discovery in application layer, so this is not gonna be good user-experience.

- i would link to the discussion https://github.com/cilium/cilium/pull/32612#discussion_r1605542681

> This method is troublesome, especially when we want to register the all Cilium nodes to a multicast group.

The method is not trouble.
I would rephrase, this is fine but does not scale when we need to use multicast with many nodes in the cluster.

> Example commands I suggest

`Suggested command examples`

> cilium multicast group 239.1.1.1 add all

if this goes to cluster-wide operation, why do we need `all`? sounds redundant? and else where.
besides, for me `cilium multicast group add <IP>` looks clear.

> cilium multicast group 239.1.1.1 delete all

can we delete all group IP addresses if IP is not specified? or single IP address is required in command line?

> cilium multicast group 239.1.1.1 list

i think most user-friendly debug command here is, `cilium multicast list --all` to dump the multicast configuration at all.
And user might want to check the specific node only, or as described specific group IP address.

> In the future, it would be beneficial to introduce commands like the following:

again, this needs to be discussed with maintainers,

> cilium multicast group 239.1.1.1 add <node-name or node IP address>

are we allow to do,

```bash
cilium multicast group 239.1.1.1 add ciliumnode-A
cilium multicast group 239.1.1.1 add ciliumnode-B
cilium multicast group 239.1.1.1 add ciliumnode-C
```

or 

```shell
cilium multicast group 239.1.1.1 add ciliumnode-A ciliumnode-B ciliumnode-C
```

for the former, we need to list 1st before the configuration.

> In cilium (NOT cilium-cli),

lets delete this section, everyone knows how this works. no need to explain.

> If this proposal is accepted, I would like to take responsibility for the implementation.

lets delete this section, this is the issue but assignment. not related to issue header.

**Comment by yushoyamaguchi** [comment]
```
# list 
## list group
cilium multicast list group
## list subscriber
cilium multicast list subscriber --all
cilium multicast list subscriber --group-ip 239.255.0.1
# add 
cilium multicast add --group-ip 239.255.0.1
# del 
cilium multicast del --group-ip 239.255.0.1
```

**Comment by yushoyamaguchi** [comment]
```
# list 
## list group
cilium multicast list group
## list subscriber
cilium multicast list subscriber --all
cilium multicast list subscriber --group-ip 239.255.0.1
# add 
cilium multicast add --group-ip 239.255.0.1
# del 
cilium multicast del --group-ip 239.255.0.1
```
